### PR TITLE
BOAC-3935: Added API for initializing degree

### PR DESCRIPTION
**Pull request by cesarvh**
https://jira.ets.berkeley.edu/jira/browse/BOAC-3935

### CODE REVIEW

**Comment by cesarvh** [comment]
Hey Anne, I made a few changes to how the Degree page is laid out. Instead of creating a separate view, I'm using conditional rendering to render different components depending on if the degree exists or if its being made.

**Comment by lyttam** [comment]
That's great @cesarvh, your single-view changes should merge nicely with what I'm currently working on for unit requirements.

File: boac/models/degree_progress_unit_requirement.py [hunk 1]
```diff
@@ -25,8 +25,6 @@
 
 from boac import db, std_commit
 from boac.models.base import Base
-from boac.models.degree_progress_template import DegreeProgressTemplate
```
Comment (lyttam) [inline boac/models/degree_progress_unit_requirement.py:28]: Did you mean to remove this import? The class is referenced on line 72.

File: boac/models/degree_progress_template.py [hunk 2]
```diff
@@ -24,11 +24,11 @@
 """
 
 from boac import db, std_commit
+from boac.models.degree_progress_unit_requirement import DegreeProgressUnitRequirement
```
Comment (lyttam) [inline boac/models/degree_progress_template.py:27]: Did you mean to add this import? The class itself isn't referenced, just the string name on line 43.

File: src/api/cohort.ts [hunk 3]
```diff
@@ -16,7 +16,8 @@ export function createCohort(
     })
     .then(response => {
       const cohort = response.data
-      store.commit('currentUserExtras/cohortCreated', cohort)
+      // TO DO: Store the degrees once we're able to retrieve them
+      // store.commit('currentUserExtras/cohortCreated', cohort)
```
Comment (lyttam) [inline src/api/cohort.ts:20]: Did you mean to put this in degree.ts instead?

File: src/views/CreateDegreeTemplate.vue [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      this.degreeId = this.$_.get(this.$route, 'params.id') || ''
+    } else {
+      this.$router.push({path: '/404'})
+    }
+  }
+}
+</script>
+
+<style>
+#label-of-create-degree-input {
```
Comment (lyttam) [inline src/views/CreateDegreeTemplate.vue:64]: I think we prefer a CSS class name here, instead of an ID.

**Pull request merged**